tests/File/CommonFileTest.php

```diff
@@ -381,112 +368,85 @@ public function testUploadAndDownloadSlicedFile(): void
-        $file = $this->_client->getFile($fileId);
+        $this->_testClient->addFileTag($fileId, 'new');
+        $file = $this->_testClient->getFile($fileId);
         $this->assertEquals(['image', 'new'], $file['tags']);
 
-        $this->_client->addFileTag($fileId, 'new');
-        $file = $this->_client->getFile($fileId);
+        $this->_testClient->addFileTag($fileId, 'new');
+        $file = $this->_testClient->getFile($fileId);
         $this->assertEquals(['image', 'new'], $file['tags'], 'duplicate tag add is ignored');
     }
 
-    public function testReadOnlyRoleFilesPermissions(): void
-    {
-        $expectedError = 'You don\'t have access to the resource.';
-        $readOnlyClient = $this->getClientForToken(STORAGE_API_READ_ONLY_TOKEN);
-
-        $options = new FileUploadOptions();
-        $fileId = $this->createAndWaitForFile(__DIR__ . '/../_data/files.upload.txt', $options);
-        $originalFile = $this->_client->getFile($fileId);
-        unset($originalFile['url']);
-
-        $filesCount = count($this->_client->listFiles(new ListFilesOptions()));
-        $this->assertGreaterThan(0, $filesCount);
-
-        $this->assertCount($filesCount, $readOnlyClient->listFiles(new ListFilesOptions()));
-
-        try {
-            $readOnlyClient->addFileTag($fileId, 'test');
-            $this->fail('Files API POST request should be restricted for readOnly user');
-        } catch (ClientException $e) {
-            $this->assertSame(403, $e->getCode());
-            $this->assertSame('accessDenied', $e->getStringCode());
-            $this->assertSame($expectedError, $e->getMessage());
-        }
-
-        try {
-            $readOnlyClient->deleteFile($fileId);
-            $this->fail('Files API DELETE request should be restricted for readOnly user');
-        } catch (ClientException $e) {
-            $this->assertSame(403, $e->getCode());
-            $this->assertSame('accessDenied', $e->getStringCode());
-            $this->assertSame($expectedError, $e->getMessage());
-        }
-
-        $file = $this->_client->getFile($fileId);
-        unset($file['url']);
-        $this->assertSame($originalFile, $file);
-    }
-
-
     /** @dataProvider invalidIdDataProvider */
-    public function testInvalidFileId($fileId): void
+    public function testInvalidFileId(string $devBranchType, string $userRole, $fileId): void
     {
         $this->expectException(ClientException::class);
         $this->expectExceptionMessage('File id cannot be empty');
-        $this->_client->getFile($fileId);
+        $this->_testClient->getFile($fileId);
     }
 
     public function invalidIdDataProvider()
     {
-        return [
-            [null],
-            [''],
+        $invalidData = [
+            'null' => [null],
+            'empty string' => [''],
         ];
+        $clientProvider = $this->provideComponentsClientTypeBasedOnSuite();
+
+        foreach ($invalidData as $desc1 => $invalid) {
```

Comment:
toto se tady ukazuje asi celkem 3x. Neslo by to predhodit tomu `provideComponentsClientTypeBasedOnSuite` (nebo nejake obalujici metode), kde by se to vyresilo obecne?